tests/test_camera.py

```diff
@@ -1,130 +1,165 @@
+
+    camera.width = 34
+    assert isclose(camera.width, 34.7826, rel_tol=0.01)
+    assert isclose(camera.height, 26.08695, rel_tol=0.01)
+
+    camera.height = 100
+    assert isclose(camera.width, 133.33333, rel_tol=0.01)
+    assert camera.height == 100
+
+
+@pytest.mark.parametrize("position, expected", [
+    [Vector(0, 0), {"right": 5, "left": -5, "top": 3.75, "bottom": -3.75}],
+    [Vector(0, 1), {"right": 5, "left": -5, "top": 4.75, "bottom": -2.75}],
+    [Vector(15, -33), {"right": 20, "left": 10, "top": -29.25, "bottom": -36.75}]
+])
+def test_camera_sides(camera, position, expected):
+    camera.position = position
+    assert camera.left == expected["left"]
+    assert camera.right == expected["right"]
+    assert camera.top == expected["top"]
+    assert camera.bottom == expected["bottom"]
+
+
+@pytest.mark.parametrize("position, expected", [
+    [Vector(0, 0), {"tl": Vector(-5, 3.75), "tr": Vector(5, 3.75), "bl": Vector(-5, -3.75), "br": Vector(5, -3.75)}],
+    [Vector(3, 6), {"tl": Vector(-2, 9.75), "tr": Vector(8, 9.75), "bl": Vector(-2, 2.25), "br": Vector(8, 2.25)}],
+    [Vector(85, -2), {"tl": Vector(80, 1.75), "tr": Vector(90, 1.75), "bl": Vector(80, -5.75), "br": Vector(90, -5.75)}]
+])
+def test_camera_corners(camera, position, expected):
+    camera.position = position
+    assert camera.top_left == expected["tl"]
+    assert camera.top_right == expected["tr"]
+    assert camera.bottom_left == expected["bl"]
+    assert camera.bottom_right == expected["br"]
+
+
+@pytest.mark.parametrize("position, point, expect", [
+    [Vector(0, 0), Vector(-4, 3), True],
+    [Vector(0, 0), Vector(-7, 1), False],
+    [Vector(50, 24), Vector(46, 26), True],
+    [Vector(50, 24), Vector(0, 0), False],
+    [Vector(0, 0), Vector(5, 3.75), True],  # Points on the edge are visible.
+    [Vector(0, 0), Vector(5, 4), False],
+    [Vector(0, 0), Vector(0, 3.75), True],
+])
+def test_camera_point_is_visible(camera, position, point, expect):
+    camera.position = position
+    assert camera.point_is_visible(point) == expect
+
+
+@pytest.mark.parametrize("position, point, expected", [
+    [Vector(0, 0), Vector(-1, -1), Vector(320, 380)],
+    [Vector(0, 0), Vector(4, -3), Vector(720, 540)],
+    [Vector(0, 0), Vector(-6, 4), Vector(-80, -20)]
+])
+def test_camera_translate_point_to_screen(camera, position, point, expected):
+    camera.position = position
+    assert camera.translate_point_to_screen(point) == expected
+
+
+@pytest.mark.parametrize("position, point, expected", [
+    [Vector(0, 0), Vector(320, 380), Vector(-1, -1)],
+    [Vector(0, 0), Vector(720, 540), Vector(4, -3)],
+    [Vector(0, 0), Vector(-80, -20), Vector(-6, 4)]
+])
+def test_camera_translate_point_to_game_space(camera, position, point, expected):
+    camera.position = position
+    assert camera.translate_point_to_game_space(point) == expected
+
+
+# @pytest.mark.skip(reason="Test for old camera. Will want to restore this functionality in new camera.")
```

Comment:
Any reason these tests weren't ported?

Reply:
Well, this top one is because I didn't rewrite `sprite_in_viewport` because it's not used by anything in the library. I'll probably do it sometime in the next couple of months and then restore/rewrite the relevant test.

The one below it is because it's doing a lot, would require translation to the new camera tests, and I'm not entirely sure what all of the steps are there to do. As such, I'd rather rewrite it as its own thing and not tied to the giant pile of other changes going on. (And it's commented out because the old camera got killed, I'd have just left it marked skip otherwise.)

Reply:
I think that's a useful utility, although the name isn't great.

The one below that is supposed to be a simple round-trip test: If we convert a point from game units to pixels and back, do we get the same point?

Reply:
It is a useful utility! I have no intention of leaving it cut forever. I just wanted to get the new camera _working_ then I'll restore some of the niceties later.

The good news RE: round tripping, the tests for translate_point_to_screen and translate_point_to_game_space use the same values for their tests, just reversing input and output.

Reply:
Well, doing a roundtrip test means it's generic and we can throw hypothesis at it.